Stack.java

```diff
@@ -0,0 +1,34 @@
+class Stack {
+
+    int[] s;
+
+    int length;
+    boolean overflow;
+    boolean underflow;
+
+    Stack(int size) {
+        s = new int[size];
+        length = 0;
+        overflow = false;
+        underflow = false;
+    }
+
+    void push(int x)   {
+        if (length >= s.length) {
+            overflow = true;
+        }
+        else {
```

Comment:
The else-condition should implement the functionality of pushing onto the stack. Simply marking the overflow as false doesn't suffice